src/plugins/kibana_react/public/code_editor/code_editor.tsx

```diff
@@ -100,118 +104,294 @@ export interface Props {
+    setIsHintActive(true);
+  }, []);
+
+  const onKeyDownHint = useCallback(
+    (ev: React.KeyboardEvent) => {
+      if (ev.key === keys.ENTER) {
+        ev.preventDefault();
+        startEditing();
       }
-
-      if (this.props.signatureProvider) {
-        monaco.languages.registerSignatureHelpProvider(
-          this.props.languageId,
-          this.props.signatureProvider
-        );
+    },
+    [startEditing]
+  );
+
+  const onKeydownMonaco = useCallback(
+    (ev: monaco.IKeyboardEvent) => {
+      if (ev.keyCode === monaco.KeyCode.Escape) {
+        // If the autocompletion context menu is open then we want to let ESCAPE close it but
+        // **not** exit out of editing mode.
+        if (!isSuggestionMenuOpen.current) {
+          ev.preventDefault();
+          ev.stopPropagation();
+          stopEditing();
+          editorHint.current?.focus();
+        }
       }
+    },
+    [stopEditing]
+  );
 
-      if (this.props.hoverProvider) {
-        monaco.languages.registerHoverProvider(this.props.languageId, this.props.hoverProvider);
-      }
+  const onBlurMonaco = useCallback(() => {
+    stopEditing();
+  }, [stopEditing]);
 
-      if (this.props.languageConfiguration) {
-        monaco.languages.setLanguageConfiguration(
-          this.props.languageId,
-          this.props.languageConfiguration
-        );
-      }
-    });
-
-    // Register themes
-    monaco.editor.defineTheme('euiColors', this.props.useDarkTheme ? DARK_THEME : LIGHT_THEME);
-    monaco.editor.defineTheme(
-      'euiColorsTransparent',
-      this.props.useDarkTheme ? DARK_THEME_TRANSPARENT : LIGHT_THEME_TRANSPARENT
+  const renderPrompt = useCallback(() => {
+    return (
+      <EuiToolTip
+        display="block"
+        content={
+          <div>
+            <EuiText>
+              <p>
+                {isReadOnly ? (
+                  <FormattedMessage
+                    id="kibana-react.kibanaCodeEditor.startEditingReadOnly"
+                    defaultMessage="Press Enter to start interacting with the code."
+                  />
+                ) : (
+                  <FormattedMessage
+                    id="kibana-react.kibanaCodeEditor.startEditing"
+                    defaultMessage="Press Enter to start editing."
+                  />
+                )}
+              </p>
+            </EuiText>
+
+            <EuiText>
+              <p>
+                {isReadOnly ? (
+                  <FormattedMessage
+                    id="kibana-react.kibanaCodeEditor.stopEditingReadOnly"
+                    defaultMessage="Press Escape to stop interacting with the code."
+                  />
+                ) : (
+                  <FormattedMessage
+                    id="kibana-react.kibanaCodeEditor.stopEditing"
+                    defaultMessage="Press Escape to stop editing."
+                  />
+                )}
+              </p>
+            </EuiText>
+          </div>
```

Comment:
The EuiTooltip already has text styles applied to it so this can be simplified to:
```suggestion
          <>
              <p>
                {isReadOnly ? (
                  <FormattedMessage
                    id="kibana-react.kibanaCodeEditor.startEditingReadOnly"
                    defaultMessage="Press Enter to start interacting with the code."
                  />
                ) : (
                  <FormattedMessage
                    id="kibana-react.kibanaCodeEditor.startEditing"
                    defaultMessage="Press Enter to start editing."
                  />
                )}
              </p>
              <p>
                {isReadOnly ? (
                  <FormattedMessage
                    id="kibana-react.kibanaCodeEditor.stopEditingReadOnly"
                    defaultMessage="Press Escape to stop interacting with the code."
                  />
                ) : (
                  <FormattedMessage
                    id="kibana-react.kibanaCodeEditor.stopEditing"
                    defaultMessage="Press Escape to stop editing."
                  />
                )}
              </p>
          </>
```

Reply:
And it looks like according to the writing guidelines the key references should be exactly whats on the keyboard, plus capitalization and bold. So it should render as:

> Press **Esc** to stop editing.

Reply:
@cchaos Could you refer me to the exact part of the writing guidelines? I am not finding it anywhere in https://elastic.github.io/eui/#/guidelines/writing

I am mainly asking because I wonder what that means for "Enter" above? I mean the key does not state "Enter" on most keyboard, but has a symbol on it, are we trying to use that symbol in that case too?

Reply:
Sorry, I had meant to link to the doc I got them from (had to ask a UX writer): https://wiki.elastic.co/display/DOC/On+text+highlighting